src/item.cpp

```diff
@@ -10243,15 +10243,24 @@ ret_val<void> item::can_contain( const item &it, const bool nested,
 
         // If the current pocket has restrictions or blacklists the item,
         // try the nested pocket regardless of whether it's soft or rigid.
+        bool is_container;
+        bool bool_contain;
```

Comment:
Why are those variables declared outside of the loop? They are not used outside of it it.

In fact, you can remove them completely:
```C++
if( !internal_it->is_container() ) {
    continue;
}
```

Reply:
It is the remains of the code for testing.
At the time of testing, I had commented out that continue and compared it at the point before returning.
Indeed it is possible to remove it as you have mentioned.
If there are no other suggestions on what to do in this area, I would like to follow your instruction.

Reply:
> In this case, does declaring inside the loop make no difference (or even improve) performance?

I don't known, you have to measure it.
Since the behaviour is the same in either case, the compiler can issue the same instructions in either case. So there might not be a difference at all.

>  I would beinitializing variable every time.
Well, the code is not initializing the variable every time, but it is assigning it every time, which is practically the same for a bool. So there is probably no difference at all.

But regarding optimization: we are talking about assignment of a single bool. The surrounding code is much more expensive, there are function calls etc. - the single bool assignment will not have a noticeable effect on runtime.

And the current code may even prevent optimization. The compiler doesn't even need a variable with the direct check of the condition `if( internal_it->is_container() ) {`

And there is the potential of a future bug when somebody uses those variables after the loop when they are potentially uninitialized.

Reply:
Thanks a lot.
I learned a lot from your advice.

> And there is the potential of a future bug when somebody uses those variables after the loop when they are potentially uninitialized.

I will be conscious of that in the future.

The code will be fixed later.